src/app/InteractionModelEngine.cpp

```diff
@@ -116,6 +117,8 @@ CHIP_ERROR InteractionModelEngine::Init(Messaging::ExchangeManager * apExchangeM
 
 void InteractionModelEngine::Shutdown()
 {
+    VerifyOrReturn(State::kInitialized == mState);
```

Comment:
So this relies on the fact that:

1. We must have failed to RegisterUnsolicitedMessageHandlerForProtocol() is state is not Initialized and
2. We are never actually doing RemoveFabricDelegate?

At the very least, this needs some comments in Init() explaining the invariants.  Better yet would be having an mState that keeps track of how far we got in Init so we can unwind from there in Shutdown....

Reply:
I updated **mState** to include the **kInitializing** state and modified **Shutdown** to check for **kUninitialized**. 
So far, I haven’t found a need to differentiate the shutdown behavior between **kInitializing** and **kInitialized**, so I’ve left it as is.